proxy.go

```diff
@@ -352,6 +353,21 @@ func realMain() {
 	var channel, proxyAddr, agentLogsSocket, logLevel string
 	var showVersion bool
 
+	// If GOMAXPROCS has not been set, restrict our thread usage
+	// so we don't grow many idle threads on large core count systems,
+	// which un-necessarily consume host PID space (and thus set an
+	// artificial max limit on the number of concurrent containers we can
+	// run)
+	if os.Getenv("GOMAXPROCS") == "" {
+		// We choose 6 as we want a couple of threads for the runtime (gc etc.)
+		// and couple of threads for our parallel user code copy routines.
+		const maxthreads = 6
```

Comment:
Might be worth putting this at the top of the file to make it stand out a little more